src/App.tsx

```diff
@@ -1,50 +1,77 @@
-import React from 'react';
 import 'bulma/css/bulma.css';
 import './App.scss';
-
-export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
+import { useState } from 'react';
+import { goodsArrayFromServer } from './types/GoodsFromServer';
+import { SortType } from './types/SortType';
 
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | ''>('');
+  const [isReverse, setIsReverse] = useState(false);
+
+  const visibleGoods = [...goodsArrayFromServer].sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.Alphabetically:
+        return good1.localeCompare(good2);
+      case SortType.Length:
+        return good1.length - good2.length;
+      default:
+        return SortType.Default;
+    }
+  });
```

Comment:
According to the checklist, interacting directly with DOM elements is discouraged in React. Here, you're manipulating the array directly which is a form of direct DOM interaction. Instead, consider using React state and effects to handle sorting and reversing of the goods array.